src/ray/stats/metric_exporter.cc

```diff
@@ -106,8 +101,38 @@ void MetricExporter::ExportViewData(
       break;
     }
   }
-  RAY_LOG(DEBUG) << "Point size : " << points.size();
-  metric_exporter_client_->ReportMetrics(points);
+
+  // TODO(simon): Batching in the exporter is a little weird. Shouldn't it be a
+  // configurable parameter in each client instead? Especially now the `points` and `data`
+  // can be different size.
```

Comment:
@ashione @rkooo567 What do you think? Currently to make test pass I have to implement the batching logic as below. But I think it makes more sense to perform batching in the client. `AgentExporterClient` and `OpenTSDBClient` and `GCSClient` each has their max batch size and they should be independent of each other.